src/System.Runtime.Analyzers/Core/SpecifyIFormatProvider.cs

```diff
@@ -27,42 +34,203 @@ public abstract class SpecifyIFormatProviderAnalyzer : DiagnosticAnalyzer
+
+                var activatorType = csaContext.Compilation.GetTypeByMetadataName("System.Activator");
+                var resourceManagerType = csaContext.Compilation.GetTypeByMetadataName("System.Resources.ResourceManager");
+
+                var computerInfoType = csaContext.Compilation.GetTypeByMetadataName("Microsoft.VisualBasic.Devices.ComputerInfo");
+                var installedUICulturePropertyOfComputerInfoType = computerInfoType?.GetMembers("InstalledUICulture").OfType<IPropertySymbol>().SingleOrDefault();
+                #endregion
+
+                csaContext.RegisterOperationAction(oaContext =>
+                {
+                    var invocationExpression = (IInvocationExpression)oaContext.Operation;
+                    var targetMethod = invocationExpression.TargetMethod;
+
+                    #region "Exceptions"
+                    if (targetMethod.IsGenericMethod ||
+                        (targetMethod.ContainingType != null && 
+                         (activatorType != null && activatorType.Equals(targetMethod.ContainingType)) ||
+                         (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType))))
+                    {
+                        return;
+                    }
+                    #endregion
+
+                    #region "IFormatProviderAlternateStringRule Only"
+                    if (stringType != null && CultureInfoType != null &&
+                        (targetMethod.Equals(stringFormatMemberWithStringAndObjectParameter) ||
+                         targetMethod.Equals(stringFormatMemberWithStringObjectAndObjectParameter) ||
+                         targetMethod.Equals(stringFormatMemberWithStringObjectObjectAndObjectParameter) ||
+                         targetMethod.Equals(stringFormatMemberWithStringAndParamsObjectParameter)))
+                    {
+                        // Sample message for IFormatProviderAlternateStringRule: Because the behavior of string.Format(string, object) could vary based on the current user's locale settings,
+                        // replace this call in IFormatProviderStringTest.M() with a call to string.Format(System.IFormatProvider, string, params object[]). If the result of 
+                        // string.Format(System.IFormatProvider, string, params object[]) will be displayed to the user, specify System.Globalization.CultureInfo.CurrentCulture as the 'IFormatProvider' parameter.
+                        // Otherwise, if the result will be stored and accessed by software, such as when it is persisted to disk or to a database, specify System.Globalization.CultureInfo.InvariantCulture.
+                        oaContext.ReportDiagnostic(
+                            invocationExpression.Syntax.CreateDiagnostic(
+                                IFormatProviderAlternateStringRule,
+                                targetMethod.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
```

Comment:
`CultureInfo` itself is not ambiguous but there are places where we mention `InstalledUICulture`, a property of `CultureInfo`. This could either be `System.Globalization.CultureInfo.InstalledUICulture` or `Microsoft.VisualBasic.Devices.ComputerInfo.InstalledUICulture`. You can checkout the tests where you will find this.